src/psyclone/psyir/backend/fortran.py

```diff
@@ -213,25 +284,26 @@ def kernelschedule_node(self, node):
         if not node.name:
             raise VisitorError("Expected node name to have a value.")
 
-        module_name = node.name.rstrip("_code") + "_mod"
-        result = (
-            "{0}module {1}\n"
-            "".format(self._nindent, module_name))
-
-        self._depth += 1
         args = [symbol.name for symbol in node.symbol_table.argument_list]
-        result += (
-            "{0}use constants_mod, only : r_def, i_def\n"
-            "{0}implicit none\n"
-            "{0}contains\n"
+        result = (
             "{0}subroutine {1}({2})\n"
             "".format(self._nindent, node.name, ",".join(args)))
 
         self._depth += 1
         # Declare the kernel data.
         declarations = ""
-        for symbol in node.symbol_table.symbols:
-            declarations += self.gen_declaration(symbol)
+        # Use statements
```

Comment:
Is this an exact duplicate of the code dealing with declarations within a Container (module)? If so, I think some re-factoring is in order.

Reply:
No, not quite. Containers do not have arguments, so it is 2/3 the same. I did think it easier to keep them separate because of this. However, I think you are right to want to have one routine, particularly if we want to control the ordering of declarations more carefully. I've therefore modified the code. This has introduced some additional tests.

Reply:
Thanks, I think it's nicer now.